app/scripts/modules/kubernetes/loadBalancer/details/details.controller.js

```diff
@@ -86,21 +87,22 @@ module.exports = angular.module('spinnaker.loadBalancer.kubernetes.details.contr
         return;
       }
 
-      var taskMonitor = {
+      const taskMonitor = {
         application: application,
         title: 'Deleting ' + loadBalancer.name,
         forceRefreshMessage: 'Refreshing application...',
         forceRefreshEnabled: true
       };
 
-      var submitMethod = function () {
-        loadBalancer.providerType = $scope.loadBalancer.provider;
-        return loadBalancerWriter.deleteLoadBalancer(loadBalancer, application, {
-          loadBalancerName: loadBalancer.name,
-          namespace: loadBalancer.region,
-        });
+      const command = {
+        cloudProvider: 'kubernetes',
+        loadBalancerName: $scope.loadBalancer.name,
+        credentials: $scope.loadBalancer.accountId,
```

Comment:
Turns out this also needs to be `$scope.loadBalancer.account`. `$scope.loadBalancer.accountId` is wrong in the object passed to `confirmationModalService.confirm`, so I'm guessing it's not needed there.